src/core_ct/core.py

```diff
@@ -52,4 +52,61 @@ def slice(self, axis, loc):
             case 2:
                 return self.pixel_array[:,:,loc]
             case _:
-                raise Exception("axis must be a value between 0 and 2 (inclusive)")
\ No newline at end of file
+                raise Exception("axis must be a value between 0 and 2 (inclusive)")
+
+    def trim(self, axis, trim_amount):
+        """Get a three-dimensional slice of the core scan trimming off the outside specified amount on the specified
+            axis.
+
+        Parameters:
+            axis -- integer either 0,1,2 specifying which dimension to collapse:
+                    0 corresponds to x-axis
+                    1 corresponds to y-axis
+                    2 corresponds to z-axis
+            loc -- integer value along the axis specifying the location of the slice
+
+        Returns:
+            3D numpy array representing the trimmed section of the core
+
+        Raises:
+            Exception if axis is a value other than 0, 1, or 2
+        """
+
+        match axis:
+            case 0:
+                self.pixel_array = self.pixel_array[trim_amount:len(self.pixel_array)-trim_amount]
+            case 1:
+                self.pixel_array = self.pixel_array[:, trim_amount:len(self.pixel_array[0])-trim_amount]
+            case 2:
+                self.pixel_array = self.pixel_array[:, :, trim_amount:len(self.pixel_array[0, 0])-trim_amount]
+            case _:
+                raise Exception("axis must be a value between 0 and 2 (inclusive)")
+
+    def chunk(self, core, x1, y1, z1, x2, y2, z2):
```

Comment:
Could you add default values to x/y/z1 and x/y/z2? Maybe None would work and we would just process that inside the function and set them to the correct constraints based on the shape of the pixel data.

Other default values could also work like just 0 for min and maybe just finding the correct shape size for the x/y/z2